nominal/core/batch_processor.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+from datetime import datetime
+from itertools import groupby
+from typing import Sequence
+
+from google.protobuf.timestamp_pb2 import Timestamp
+from nominal_api_protos.nominal_write_pb2 import (
+    Channel as NominalChannel,
+)
+from nominal_api_protos.nominal_write_pb2 import (
+    DoublePoint,
+    DoublePoints,
+    Points,
+    Series,
+    StringPoint,
+    StringPoints,
+    WriteRequestNominal,
+)
+
+from nominal.core._clientsbunch import ProtoWriteService
+from nominal.core.stream import BatchItem
+from nominal.ts import IntegralNanosecondsUTC, _SecondsNanos
+
+
+def _to_api_batch_key(item: BatchItem) -> tuple[str, Sequence[tuple[str, str]], str]:
+    return item.channel_name, sorted(item.tags.items()) if item.tags is not None else [], type(item.value).__name__
+
+
+def make_points_proto(api_batch: Sequence[BatchItem]) -> Points:
+    # Check first value to determine type
+    sample_value = api_batch[0].value
```

Comment:
we should def clean up these types in the future; there should be zero need to peek and test as we know the types ahead of time (i.e. at the point of enqueueing)